dat/generated_tables.py

```diff
@@ -0,0 +1,73 @@
+from typing import Sequence
+
+from dat.model.row_collections import RowCollection
+from dat.model.table import GeneratedReferenceTable, ReferenceTable
+
+reference_table_1 = GeneratedReferenceTable(
+    table_name='reference_table_1',
+    table_description='My first table',
+    column_names=['letter', 'number', 'a_float'],
+    partition_keys=['letter'],
+    reader_protocol_version=2,
```

Comment:
I am not a maintainer of Spark, but I would be surprised if the Delta jar would be bundled within the standard Spark distribution, this would be constraining (if you want to use Spark x, you need to use Delta y).

I would expect that when you run Spark, you need to specify some configuration settings that tells Delta is enabled and place Delta in the classpath of the application. I did the following with spark-xml in a project:

```
@pytest.fixture(scope='session')
def spark_session(request):
    session_builder = SparkSession.builder.appName(
        'My Project Automated Testing'
    ).config(
        'spark.jars',
        build_sparkxml_jar_location()
    ).config(
        'spark.executorEnv.MLFLOW_TRACKING_URI',
        mlflow_helper.mlflow_test_tracking_uri
    )
  spark = session_builder.getOrCreate()
  request.addfinalizer(spark.stop)
  return spark
```

Databricks team, can you confirm?

Reply:
Ah sorry for confusion. When I used Spark more I was using Databricks version, which does bundle Delta Lake.

I just want to know what protocol versions are being written out by default in the latest Delta jar right now, so we know which are most prevalent in the real world.

Reply:
It is documented as a part of the databricks runtime, for example if you click here: https://docs.databricks.com/release-notes/runtime/11.1.html

you can see in the section "System environment" Delta Lake: 2.0.0

Reply:
That is the library version, not the protocol version.